aana/deployments/task_queue_deployment.py

```diff
@@ -61,81 +58,82 @@ async def apply_config(self, config: dict[str, Any]):
-                > aana_settings.task_queue.num_workers * 2
-            )
+        running_task_ids: list[str] = []
+        handle = None
+
+        active_tasks = self.task_repo.get_active_tasks()
+        for task in active_tasks:
+            if task.status == TaskStatus.RUNNING:
+                running_task_ids.append(task.id)
+            if task.status == TaskStatus.ASSIGNED:
+                self.task_repo.update_status(
+                    task_id=task.id,
+                    status=TaskStatus.NOT_FINISHED,
+                    progress=0,
+                )
 
         while True:
             if not self._configured:
                 # Wait for the deployment to be configured.
                 await asyncio.sleep(1)
                 continue
 
-            # Remove completed tasks from the futures dictionary
-            for task_id in list(self.futures.keys()):
-                if self.futures[task_id].done():
-                    del self.futures[task_id]
-
-            if is_thread_pool_full():
-                # wait a bit to give the thread pool time to process tasks
+            # Remove completed tasks from the list of running tasks
+            running_task_ids = self.task_repo.remove_completed_tasks(running_task_ids)
+
+            # Check for expired tasks
+            execution_timeout = aana_settings.task_queue.execution_timeout
+            expired_tasks = self.task_repo.get_expired_tasks(execution_timeout)
+            for task in expired_tasks:
+                if task.num_retries >= aana_settings.task_queue.max_retries:
+                    self.task_repo.update_status(
+                        task_id=task.id,
+                        status=TaskStatus.FAILED,
+                        progress=0,
+                        result={
+                            "error": "TimeoutError",
+                            "message": (
+                                f"Task execution timed out after {execution_timeout} seconds and "
+                                f"exceeded the maximum number of retries ({aana_settings.task_queue.max_retries})"
+                            ),
+                        },
+                    )
+                else:
+                    self.task_repo.update_status(
+                        task_id=task.id,
+                        status=TaskStatus.ASSIGNED,
+                        progress=0,
+                    )
+
+            # If the queue is full, wait and retry
+            if len(running_task_ids) >= aana_settings.task_queue.num_workers:
                 await asyncio.sleep(0.1)
                 continue
 
-            tasks = self.task_repo.get_unprocessed_tasks(
-                limit=aana_settings.task_queue.num_workers * 2
+            # Get new tasks from the database
+            num_tasks_to_assign = aana_settings.task_queue.num_workers - len(
+                running_task_ids
             )
+            tasks = self.task_repo.get_unprocessed_tasks(limit=num_tasks_to_assign)
```

Comment:
Where do you get the assing task for executing them again?
Looks like the assigned task (cauesed by expircy) will noy be submited again anywhere. We just increase the num_retries for them.